src/main/java/com/conveyal/datatools/manager/utils/MergeFeedUtils.java

```diff
@@ -0,0 +1,225 @@
+    /** Checks that any of a set of errors is of the type {@link NewGTFSErrorType#DUPLICATE_ID}. */
+    public static boolean hasDuplicateError(Set<NewGTFSError> errors) {
+        for (NewGTFSError error : errors) {
+            if (error.errorType.equals(NewGTFSErrorType.DUPLICATE_ID)) return true;
+        }
+        return false;
+    }
+
+    /** Get table-scoped value used for key when remapping references for a particular feed. */
+    public static String getTableScopedValue(Table table, String prefix, String id) {
+        return String.join(":",
+            table.name,
+            prefix,
+            id);
+    }
+
+    /**
+     * Checks whether the future and active stop_times for a particular trip_id are an exact match.
+     */
+    public static boolean stopTimesMatch(List<StopTime> futureStopTimes, List<StopTime> activeStopTimes) {
+        if (futureStopTimes.size() != activeStopTimes.size()) {
+            return false;
+        }
+        for (int i = 0; i < activeStopTimes.size(); i++) {
+            if (!activeStopTimes.get(i).equals(futureStopTimes.get(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Get parent feed source depending on merge type. Assign regional feed source or simply the first parent feed
+     * source found in the feed version list (these should all belong to the same feed source if the merge is not
+     * regional).
+     */
+    public static FeedSource getParentFeedSourceForMerge(MergeFeedsJob job, boolean storeNewVersion) {
+        FeedSource regionalFeedSource = null;
+        Project project = Persistence.projects.getById(job.projectId);
+        // If storing a regional merge as a new version, find the feed source designated by the project.
+        if (job.mergeType.equals(REGIONAL) && storeNewVersion) {
+            regionalFeedSource = Persistence.feedSources.getById(project.regionalFeedSourceId);
+            // Create new feed source if this is the first regional merge.
+            if (regionalFeedSource == null) {
+                regionalFeedSource = new FeedSource("REGIONAL MERGE", project.id, REGIONAL_MERGE);
+                // Store new feed source.
+                Persistence.feedSources.create(regionalFeedSource);
+                // Update regional feed source ID on project.
+                project.regionalFeedSourceId = regionalFeedSource.id;
+                Persistence.projects.replace(project.id, project);
+            }
+        }
+        return job.mergeType.equals(REGIONAL)
+            ? regionalFeedSource
+            : job.getFeedVersions().iterator().next().parentFeedSource();
+    }
+
+    public static FeedVersion getMergedVersion(MergeFeedsJob job, boolean storeNewVersion) {
+        FeedSource feedSource = getParentFeedSourceForMerge(job, storeNewVersion);
+        FeedRetrievalMethod retrievalMethod = job.mergeType.equals(REGIONAL)
+            ? REGIONAL_MERGE
+            : SERVICE_PERIOD_MERGE;
```

Comment:
In seeing how often there are branches for REGIONAL vs SERVICE_PERIOD, it seems like this is also another candidate for having some kind of abstract class or interface with extending/implementing classes that can do certain things based on whether the merge type is regional or service period.